atomicapp/providers/docker.py

```diff
@@ -51,15 +60,41 @@ def init(self):
             raise ProviderFailedException(msg)
 
     def deploy(self):
+        logger.info("Deploying to provider: Docker")
+
         for artifact in self.artifacts:
             artifact_path = os.path.join(self.path, artifact)
             label_run = None
             with open(artifact_path, "r") as fp:
                 label_run = fp.read().strip()
+            labels = label_run.split()
 
-            cmd = label_run.split()
+            # If --name is provided, do not re-name due to potential linking of containers. Warn user instead.
+            # Else use namespace provided within answers.conf
+            if '--name' in labels:
+                logger.info("WARNING: Using --name provided within artifact file. You will be unable to undeploy via atomicapp.")
+            else:
+                labels.insert(labels.index('run') + 1, "--name=%s_%s_%s" % (self.default_name, self.namespace, Utils.getUniqueUUID()))
 
+            cmd = labels
             if self.dryrun:
                 logger.info("DRY-RUN: %s", " ".join(cmd))
             else:
                 subprocess.check_call(cmd)
+
+    def undeploy(self):
+        logger.info("Undeploying to provider: Docker")
+
+        docker_cmd = 'docker inspect --format="{{ .Name }}" $(sudo docker ps -aq --no-trunc) | sed "s,/,,g"'
+        containers = dict((line, 1) for line in subprocess.check_output(docker_cmd, shell=True).splitlines())
+
+        # Regex checks for matching container name and kills it. ex. atomic_mariadb-atomicapp-app_9dfb369ed2a0
+        for container in containers:
+            m = re.match("%s_+%s+_+[a-zA-Z0-9]{12}" % (self.default_name, self.namespace), container)
```

Comment:
I am missing a code where `self.default_name` will get the value of `--name` if it's in the run command. I thought we agreed that we can do the undeployment by parsing this name from run command and using it in both deploy and undeploy (?), but maybe I misunderstood:-)